src/card.rs

```diff
@@ -117,57 +117,53 @@ impl fmt::Display for CardShape {
 fn random_card_number() -> Result<CardNumber, String>  {
     let index = random::<u32>() % 3;
 
-    if CardNumber::One as u32 == index {
-        return Ok(CardNumber::One);
-    } else if CardNumber::Two as u32 == index {
-        return Ok(CardNumber::Two);
-    } else if CardNumber::Three as u32 == index {
-        return Ok(CardNumber::Three);
-    } else {
-        return Err(format!("No valid card number found for {}", index));
+    match index {
+        _i if CardNumber::One as u32 == index => return Ok(CardNumber::One),
+        _i if CardNumber::Two as u32 == index => return Ok(CardNumber::Two),
+        _i if CardNumber::Three as u32 == index => return Ok(CardNumber::Three),
+        _ => (),
     }
+
+    return Err(format!("No valid card number found for {}", index));
```

Comment:
I mean, instead of one big if block it's a bunch of small ones :) But `match` is indeed good, so I think I'd do it like this, if I change it:

```suggestion
    return match index {
        _i if CardNumber::One as u32 == index => Ok(CardNumber::One),
        _i if CardNumber::Two as u32 == index => Ok(CardNumber::Two),
        _i if CardNumber::Three as u32 == index => Ok(CardNumber::Three),
        _ => Err(format!("No valid card number found for {}", index));,
    }
```

There might be a typo in there, but the gist of it is to use `match` like an expression and just return whatever it does. Hm?